test/functional/context.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2015 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    patch,
+    elementOpen,
+    elementOpenStart,
+    elementOpenEnd,
+    elementClose,
+    elementVoid,
+    currentElement
+} from '../../index';
+
+describe('context', () => {
+  var container;
+  var sandbox = sinon.sandbox.create();
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    document.body.removeChild(container);
+  });
+
+  describe('should return currentElement', () => {
+    var el
+    beforeEach(() => {
+      patch(container, () => {});
+      el = null;
+    });
+
+    it('from elementOpen', () => {
```

Comment:
its should look something like:

`it('should return the element from elementOpen', ...)`

describes should describe the feature

`describe('currentElement', ...`

Ideally, the full thing would read as a sentence, for example in element_creation.js:

"element creation when creating a single node should render with the specified tag"

Also, context isn't really a feature as far as a consumer of the library cares. The feature they deal with is just `currentElement`.